src/java/org/apache/cassandra/service/accord/AccordConfigurationService.java

```diff
@@ -399,32 +398,30 @@ private void fetchTopologyAsync(long epoch, BiConsumer<Object, ? super Throwable
                 return;
             }
 
-            try
+            Set<InetAddressAndPort> peers = new HashSet<>(metadata.directory.allJoinedEndpoints());
+            peers.remove(FBUtilities.getBroadcastAddressAndPort());
+            if (peers.isEmpty())
             {
-                Set<InetAddressAndPort> peers = new HashSet<>(metadata.directory.allJoinedEndpoints());
-                peers.remove(FBUtilities.getBroadcastAddressAndPort());
-                if (peers.isEmpty())
-                {
-                    onResult.accept(Success, null);
-                    return;
-                }
-
-                // Fetching only one epoch here since later epochs might have already been requested concurrently
-                TopologyRange result = FetchTopologies.fetch(SharedContext.Global.instance, peers, epoch, epoch).get();
-                result.forEach(this::reportTopology, epoch, 1);
                 onResult.accept(Success, null);
+                return;
             }
-            catch (Throwable e)
-            {
-                if (currentEpoch() >= epoch)
-                {
-                    onResult.accept(Success, null);
-                    return;
-                }
-                if (e instanceof InterruptedException)
-                    Thread.currentThread().interrupt();
-                onResult.accept(null, e);
-            }
+
+            // Fetching only one epoch here since later epochs might have already been requested concurrently
+            FetchTopologies.fetch(SharedContext.Global.instance, peers, epoch, epoch)
```

Comment:
This is just making `fetch` asynchronous; I think it should have no positive or negative impact on tests, but blocking `get` was unnerving, and we would get stuck here often on shutdown.